controller/watch/dynamic_resource_client.go

```diff
@@ -0,0 +1,23 @@
+package watch
```

Comment:
Can you add some comments as to why we need this? It appears to be adding a level of indirection but I'm not sure why. Maybe it'll be clearer as by the end of the PR but as a stand-alone file it might be to explain what's going on.